addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/Spring4ShellScanRuleUnitTest.java

```diff
@@ -0,0 +1,268 @@
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.NanoHTTPD.Response;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link Spring4ShellScanRule}. */
+class Spring4ShellScanRuleUnitTest extends ActiveScannerTest<Spring4ShellScanRule> {
+
+    @Override
+    protected Spring4ShellScanRule createScanner() {
+        return new Spring4ShellScanRule();
+    }
+
+    @Test
+    void shouldTargetSpringTech() {
+        // Given
+        TechSet techSet = techSet(Tech.SPRING);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotTargetNonSpringTechs() {
+        // Given
+        TechSet techSet = techSetWithout(Tech.SPRING);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(false)));
+    }
+
+    @Test
+    void shouldBeInjectionGatherAlert() {
+        // Given
+        Integer category = Category.INJECTION;
+        // When
+        Integer ruleCategory = rule.getCategory();
+        // Then
+        assertEquals(ruleCategory, category);
+    }
+
+    @Test
+    void shouldIgnore400ResponseCodes() throws HttpMalformedHeaderException {
+        // Given
+        String path = "/Ignore400";
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(Response.Status.OK, NanoHTTPD.MIME_HTML, "");
```

Comment:
Have the handler set 400 instead of ok? So that you don't have to set it below?

Reply:
No, this is the real response from the server, the other one is the original response.